File: app/src/main/java/com/cyberIyke/allvideodowloader/database/ShortcutDao.java
package com.cyberIyke.allvideodowloader.database;

import androidx.lifecycle.LiveData;
import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import java.util.List;

@Dao
public interface ShortcutDao {
    @Insert
    void insert(ShortcutTable shortcutTable);

    @Delete
    void delete(ShortcutTable shortcutTable);

    @Query("SELECT * FROM ShortcutTable")
    LiveData<List<ShortcutTable>> getAllShortcut();

    @Query("SELECT * FROM ShortcutTable")
    List<ShortcutTable> getAllShortcutList();
}

File: app/src/main/java/com/cyberIyke/allvideodowloader/database/ShortcutDao.kt
package com.cyberIyke.allvideodowloader.database

import androidx.lifecycle.LiveData
import androidx.room.*
import com.cyberIyke.allvideodowloader.database.ShortcutTable

@Dao
interface ShortcutDao {
    @Insert
    fun insert(shortcutTable: ShortcutTable?)

    @Delete
    fun delete(shortcutTable: ShortcutTable?)

    @get:Query("SELECT * FROM ShortcutTable")
    val allShortcut: LiveData<List<ShortcutTable?>?>?

    @get:Query("SELECT * FROM ShortcutTable")
    val allShortcutList: List<ShortcutTable?>?
}